File: nabcc-to-unicode-braille.pl
# nabcc-to-unicode-braille.pl
# .bseや.brlなどのNABCCのファイルをUnicode点字に変換
# 使い方:
# .bseや.brlのファイルを、utf8のテキストファイルに変換する。
# 改行コードはUnix式に変換する。
# ファイル名をsource.txtにする。
# その後、nabcc-to-unicode-braille.plを引数無しで実行する
# 変換結果は result.txt に保存される

#!/usr/bin/perl
use strict;
use warnings;
use open IN  => ":utf8";
use open OUT => ":utf8";

open(my $SOURCEFILE, '<:utf8', './source.txt') or die "Cannot open source file: $!";
open(my $RESULTFILE, '>:utf8', './result.txt') or die "Cannot open result file: $!";

while (my $line = <$SOURCEFILE>) {
    $line =~ s/A/\N{U+2801}/g;
    $line =~ s/B/\N{U+2803}/g;
    $line =~ s/C/\N{U+2809}/g;
    $line =~ s/D/\N{U+2819}/g;
    $line =~ s/E/\N{U+2811}/g;
    $line =~ s/F/\N{U+280B}/g;
    $line =~ s/G/\N{U+281B}/g;
    $line =~ s/H/\N{U+2813}/g;
    $line =~ s/I/\N{U+280A}/g;
    $line =~ s/J/\N{U+281A}/g;
    $line =~ s/K/\N{U+2805}/g;
    $line =~ s/L/\N{U+2807}/g;
    $line =~ s/M/\N{U+280D}/g;
    $line =~ s/N/\N{U+281D}/g;
    $line =~ s/O/\N{U+2815}/g;
    $line =~ s/P/\N{U+280F}/g;
    $line =~ s/Q/\N{U+281F}/g;
    $line =~ s/R/\N{U+2817}/g;
    $line =~ s/S/\N{U+280E}/g;
    $line =~ s/T/\N{U+281E}/g;
    $line =~ s/U/\N{U+2825}/g;
    $line =~ s/V/\N{U+2827}/g;
    $line =~ s/X/\N{U+282D}/g;
    $line =~ s/Y/\N{U+283D}/g;
    $line =~ s/Z/\N{U+2835}/g;
    $line =~ s/&/\N{U+282F}/g;
    $line =~ s/=/\N{U+283F}/g;
    $line =~ s/\(/\N{U+2837}/g;
    $line =~ s/\!/\N{U+282E}/g;
    $line =~ s/\)/\N{U+283E}/g;
    $line =~ s/\*/\N{U+2821}/g;
    $line =~ s/</\N{U+2823}/g;
    $line =~ s/%/\N{U+2829}/g;
    $line =~ s/\?/\N{U+2839}/g;
    $line =~ s/:/\N{U+2831}/g;
    $line =~ s/\$/\N{U+282B}/g;
    $line =~ s/\]/\N{U+283B}/g;
    $line =~ s/\\/\N{U+2833}/g;
    $line =~ s/\[/\N{U+282A}/g;
    $line =~ s/W/\N{U+283A}/g;
    $line =~ s/1/\N{U+2802}/g;
    $line =~ s/2/\N{U+2806}/g;
    $line =~ s/3/\N{U+2812}/g;
    $line =~ s/4/\N{U+2832}/g;
    $line =~ s/5/\N{U+2822}/g;
    $line =~ s/6/\N{U+2816}/g;
    $line =~ s/7/\N{U+2836}/g;
    $line =~ s/8/\N{U+2826}/g;
    $line =~ s/9/\N{U+2814}/g;
    $line =~ s/0/\N{U+2834}/g;
    $line =~ s/\//\N{U+280C}/g;
    $line =~ s/\+/\N{U+282C}/g;
    $line =~ s/>/\N{U+281C}/g;
    $line =~ s/#/\N{U+283C}/g;
    $line =~ s/-/\N{U+2824}/g;
    $line =~ s/'/\N{U+2804}/g;
    $line =~ s/@/\N{U+2808}/g;
    $line =~ s/\^/\N{U+2818}/g;
    $line =~ s/\_/\N{U+2838}/g;
    $line =~ s/"/\N{U+2810}/g;
    $line =~ s/\./\N{U+2828}/g;
    $line =~ s/;/\N{U+2830}/g;
    $line =~ s/,/\N{U+2820}/g;

    print $RESULTFILE $line;
}

close $SOURCEFILE;
close $RESULTFILE;

File: nabcc-to-unicode-braille.py
# nabcc-to-unicode-braille.py
# .bseや.brlなどのNABCCのファイルをUnicode点字に変換
# 使い方:
# python nabcc-to-unicode-braille.py 入力ファイル 出力ファイル

import sys

NABCC_TABLE = str.maketrans({
    'A': '\u2801', 'B': '\u2803', 'C': '\u2809', 'D': '\u2819',
    'E': '\u2811', 'F': '\u280B', 'G': '\u281B', 'H': '\u2813',
    'I': '\u280A', 'J': '\u281A', 'K': '\u2805', 'L': '\u2807',
    'M': '\u280D', 'N': '\u281D', 'O': '\u2815', 'P': '\u280F',
    'Q': '\u281F', 'R': '\u2817', 'S': '\u280E', 'T': '\u281E',
    'U': '\u2825', 'V': '\u2827', 'W': '\u283A', 'X': '\u282D',
    'Y': '\u283D', 'Z': '\u2835',
    '&': '\u282F', '=': '\u283F', '(': '\u2837', '!': '\u282E',
    ')': '\u283E', '*': '\u2821', '<': '\u2823', '%': '\u2829',
    '?': '\u2839', ':': '\u2831', '$': '\u282B', ']': '\u283B',
    '\\': '\u2833', '[': '\u282A',
    '1': '\u2802', '2': '\u2806', '3': '\u2812', '4': '\u2832',
    '5': '\u2822', '6': '\u2816', '7': '\u2836', '8': '\u2826',
    '9': '\u2814', '0': '\u2834',
    '/': '\u280C', '+': '\u282C', '>': '\u281C', '#': '\u283C',
    '-': '\u2824', "'": '\u2804', '@': '\u2808', '^': '\u2818',
    '_': '\u2838', '"': '\u2810', '.': '\u2828', ';': '\u2830',
    ',': '\u2820',
})

if len(sys.argv) != 3:
    print(f"使い方: python {sys.argv[0]} 入力ファイル 出力ファイル")
    sys.exit(1)

input_file = sys.argv[1]
output_file = sys.argv[2]

try:
    with open(input_file, 'r', encoding='utf-8') as src, \
         open(output_file, 'w', encoding='utf-8') as dst:
        for line in src:
            dst.write(line.translate(NABCC_TABLE))
    print(f"変換完了: {input_file} -> {output_file}")
except FileNotFoundError as e:
    print(f"エラー: ファイルが見つかりません: {e}")
    sys.exit(1)
